### ISSUE TITLE
Benchmark ignores binding redirects

### ISSUE THREAD

**Issue by Horusiath** [PerfDotNet/BenchmarkDotNet#214]
I've found an issue in the following scenario: I have a library build as CLR F# class library - it's targeting FSharp.Core lib version=4.4.0.0. However it has dependency on other lib which expects FSharp.Core version=4.3.0.0. 

Hopefully this should be solved by specifying following app.config in benchmarking console app (which is using FSharp.Core 4.4.0.0 too):

```xml
<?xml version="1.0" encoding="utf-8"?>
<configuration>
  <runtime>
    <assemblyBinding xmlns="urn:schemas-microsoft-com:asm.v1">
      <dependentAssembly>
        <assemblyIdentity name="FSharp.Core" publicKeyToken="b03f5f7f11d50a3a" culture="neutral" />
        <bindingRedirect oldVersion="0.0.0.0-4.3.1.0" newVersion="4.3.1.0" />
      </dependentAssembly>
    </assemblyBinding>
  </runtime>
</configuration>
```

However after running benchmarking application, I'm getting series of errors:

```

// *** Build ***
// Result = Success

// *** Execute ***
// Launch: 1
System.TypeInitializationException: The type initializer for '<StartupCode$FSharp-Data-GraphQL-Benchmarks>.$FSharp.Data.GraphQL.ExecutionBenchmark' threw an exception. 
   ---> System.IO.FileNotFoundException: Could not load file or assembly 'FSharp.Core, Version=4.3.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' or one of its dependencies. The system cannot find the file specified.
   at FSharp.Data.GraphQL.Parser.parse(String query)
   at <StartupCode$FSharp-Data-GraphQL-Benchmarks>.$FSharp.Data.GraphQL.ExecutionBenchmark..cctor() in D:\projects\dev\contrib\FSharp.Data.GraphQL\tests\FSharp.Data.GraphQL.Benchmarks\ExecutionBenchmark.fs:line 64
   --- End of inner exception stack trace ---
   at FSharp.Data.GraphQL.ExecutionBenchmark.SimpleExecutionBenchmark.BenchmarkSimpleQueryParsed() in D:\projects\dev\contrib\FSharp.Data.GraphQL\tests\FSharp.Data.GraphQL.Benchmarks\ExecutionBenchmark.fs:line 101
   at BenchmarkDotNet.Autogenerated.Program.Main(String[] args) in d:\projects\dev\contrib\FSharp.Data.GraphQL\tests\FSharp.Data.GraphQL.Benchmarks\bin\Release\BenchmarkDotNet.Artifacts\bin\BDN.Auto\Program.notcs:line 30
No more Benchmark runs will be launched as NO measurements were obtained from the previous run!

// Run, Diagnostic
System.TypeInitializationException: The type initializer for '<StartupCode$FSharp-Data-GraphQL-Benchmarks>.$FSharp.Data.GraphQL.ExecutionBenchmark' threw an exception. 
   ---> System.IO.FileNotFoundException: Could not load file or assembly 'FSharp.Core, Version=4.3.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' or one of its dependencies. The system cannot find the file specified.
   at FSharp.Data.GraphQL.Parser.parse(String query)
   at <StartupCode$FSharp-Data-GraphQL-Benchmarks>.$FSharp.Data.GraphQL.ExecutionBenchmark..cctor() in D:\projects\dev\contrib\FSharp.Data.GraphQL\tests\FSharp.Data.GraphQL.Benchmarks\ExecutionBenchmark.fs:line 64
   --- End of inner exception stack trace ---
   at FSharp.Data.GraphQL.ExecutionBenchmark.SimpleExecutionBenchmark.BenchmarkSimpleQueryParsed() in D:\projects\dev\contrib\FSharp.Data.GraphQL\tests\FSharp.Data.GraphQL.Benchmarks\ExecutionBenchmark.fs:line 101
   at BenchmarkDotNet.Autogenerated.Program.Main(String[] args) in d:\projects\dev\contrib\FSharp.Data.GraphQL\tests\FSharp.Data.GraphQL.Benchmarks\bin\Release\BenchmarkDotNet.Artifacts\bin\BDN.Auto\Program.notcs:line 30
```

It looks like BenchmarkDotNet is ignoring assembly binding redirects for conflicting library dependencies.

**Comment by adamsitnik** [comment]
@Horusiath Hej! Thanks for the report. The bug has been fixed few days ago, will be part of 0.9.8 release (#108)

**Comment by kolektiv** [comment]
Incidentally I think this is the same as #67 - where I've included a hopefully simple repo which can be used to reproduce if that's useful.
